src/loops/simple_programs_with_repitition/hard/q4/Reduceros.java

```diff
@@ -0,0 +1,61 @@
+package loops.simple_programs_with_repitition.hard.q4;
+
+/*
+Write a program to display a "single-digit reduction" of a number taken from the user by adding up its digits until
+a single-digit number is obtained. Eg. 567 becomes 5 + 6 + 7 = 18 = 1 + 8 => 9)
+
+Sample Output:
+Enter a number:
+567
+567's single number reduction is 9
+ */
+
+import java.util.Scanner;
+
+public class Reduceros {
+    public static void main(String[] args) {
+        Scanner input = new Scanner(System.in);
+        System.out.println("Enter a number: ");
+        int n = input.nextInt();
+        int nn = n;
+        while (n >= 10) {
+            String s = String.valueOf(n);
+            n = 0;
+            for (int x = 0; x < s.length(); x++) {
+                n += Integer.parseInt(String.valueOf(s.charAt(x)));
+            }
+        }
+        System.out.println(nn + "'s single number reduction is " + n);
+    }
+}
+
+/*
```

Comment:
I think we should have the original solution code be the one this question runs off of.